pkg/cloudprovider/providers/gce/gce_urlmap.go

```diff
@@ -29,47 +33,35 @@ func newUrlMapMetricContext(request string) *metricContext {
 // GetUrlMap returns the UrlMap by name.
 func (gce *GCECloud) GetUrlMap(name string) (*compute.UrlMap, error) {
 	mc := newUrlMapMetricContext("get")
-	v, err := gce.service.UrlMaps.Get(gce.projectID, name).Do()
+	v, err := gce.c.UrlMaps().Get(context.Background(), meta.GlobalKey(name))
 	return v, mc.Observe(err)
 }
 
 // CreateUrlMap creates a url map
 func (gce *GCECloud) CreateUrlMap(urlMap *compute.UrlMap) error {
 	mc := newUrlMapMetricContext("create")
-	op, err := gce.service.UrlMaps.Insert(gce.projectID, urlMap).Do()
-	if err != nil {
-		return mc.Observe(err)
-	}
-	return gce.waitForGlobalOp(op, mc)
+	return mc.Observe(gce.c.UrlMaps().Insert(context.Background(), meta.GlobalKey(urlMap.Name), urlMap))
 }
 
 // UpdateUrlMap applies the given UrlMap as an update
 func (gce *GCECloud) UpdateUrlMap(urlMap *compute.UrlMap) error {
 	mc := newUrlMapMetricContext("update")
-	op, err := gce.service.UrlMaps.Update(gce.projectID, urlMap.Name, urlMap).Do()
-	if err != nil {
-		return mc.Observe(err)
-	}
-	return gce.waitForGlobalOp(op, mc)
+	return mc.Observe(gce.c.UrlMaps().Update(context.Background(), meta.GlobalKey(urlMap.Name), urlMap))
 }
 
 // DeleteUrlMap deletes a url map by name.
 func (gce *GCECloud) DeleteUrlMap(name string) error {
 	mc := newUrlMapMetricContext("delete")
-	op, err := gce.service.UrlMaps.Delete(gce.projectID, name).Do()
-	if err != nil {
-		if isHTTPErrorCode(err, http.StatusNotFound) {
-			return nil
-		}
-		return mc.Observe(err)
+	err := gce.c.UrlMaps().Delete(context.Background(), meta.GlobalKey(name))
+	if isHTTPErrorCode(err, http.StatusNotFound) {
+		err = nil
```

Comment:
Remove condition. Both test/ingress-utils and ingress-gce check for NotFound.

Reply:
done